content/chainguard/migration/migration-tips.md

```diff
@@ -0,0 +1,244 @@
+
+```bash
+docker run -it cgr.dev/chainguard/python:latest-dev echo "in a shell"
+/usr/bin/python: can't open file '//echo': [Errno 2] No such file or directory
+```
+
+Other images, such as our [WordPress container images](https://images.chainguard.dev/directory/image/wordpress/overview?utm_source=cg-academy&utm_medium=referral&utm_campaign=dev-enablement&utm_content=edu-content-chainguard-migration-migrations-overview), will have a different entrypoint behavior in their `-dev` variant to allow for customization and to facilitate migration from other base images. It's important to always read the image's documentation to understand how the entrypoint works, and if there are any major differences from other images you may be used to work with.
+
+## Containers Don't Run As root By Default
+
+Although there are exceptions, Chainguard Containers typically don’t run as the root user. The reason for this is that distroless containers should have no privileged capabilities, and containers that run as a non-root user and use a minimal seccomp profile are ideal from a security perspective.
+
+Because they don't run as the root user, you may need to include a `USER root` statement in your Dockerfile before installing software on a Chainguard Container.
+
+Additionally, be aware that `-dev` images also do not run as root in most cases, which can result in permission errors like the following:
+
+```bash
+docker run -it --entrypoint bin/bash chainguard/python:latest-dev
+
+bash-5.2$ mkdir test
+mkdir: can't create directory 'test': Permission denied
+
+bash-5.2$ sudo mkdir test
+bash: sudo: command not found
+```
+
+In cases like this, you can instead run a command like the following to access the container's shell as the root user:
+
+```bash
+docker run -it --user root --entrypoint bin/bash chainguard/python:latest-dev
+```
+
+Here, the `--user` option tells Docker to assume the root user role.
+
+
+## Packages Not Found
+
+Container images are usually meant to support every possible use case. Because of this, they often contain packages that aren't necessary for many use cases, which increases the container image's attack surface and makes it more likely to contain CVEs.
+
+Chainguard Containers are built with minimalism in mind, and thus contain the bare minimum packages needed for an image to function. However, this also means that Chainguard Containers may not contain the packages that you'd expect to find in third-party alternatives.
+
+If a Chainguard Container is missing certain packages that are required for your application, we recommend using a base image and installing the required dependencies on top of it, preferably in a multi-stage Docker build. Our guides on [How to Use Chainguard Containers](/chainguard/chainguard-images/how-to-use-chainguard-images/#extending-chainguard-base-images) and [Getting Started with Distroless](/chainguard/migration/migrations-overview/) include guidance on how you can extend Chainguard base images.
```

Comment:
should mention custom assembly and private apk repos

Reply:
good call